src/plugins/multiline/index.ts

```diff
@@ -0,0 +1,393 @@
+        !this.settings.automaticallyMultilinify
+      )
+        return;
+
+      this.dequeueAllMultilinifications();
+    }, 0);
+
+    this.dispatcherID = Calc.controller.dispatcher.register((e) => {
+      if (
+        e.type === "set-item-latex" ||
+        e.type === "undo" ||
+        e.type === "redo"
+      ) {
+        this.lastEditTime = Date.now();
+      }
+
+      if (
+        e.type === "set-item-latex" ||
+        e.type === "undo" ||
+        e.type === "redo" ||
+        e.type === "tick" ||
+        e.type === "tick-ticker"
+      ) {
+        this.multilineExpressions(e);
+      }
+    });
+
+    this.customDispatcherID = registerCustomDispatchOverridingHandler((evt) => {
+      if (evt.type === "on-special-key-pressed") {
+        if (evt.key === "Up" || evt.key === "Down") {
+          if (!this.doMultilineVerticalNav(evt.key)) return false;
+        }
+      }
+    }, 0);
+  }
+
+  afterDisable() {
+    document.removeEventListener("keydown", this.keydownHandler);
+    document.removeEventListener("mousedown", this.mousedownHandler);
+    document.removeEventListener("resize", this.resizeHandler);
+
+    this.unmultilineExpressions();
+    document.body.classList.remove("multiline-expression-enabled");
+
+    if (this.dispatcherID)
+      Calc.controller.dispatcher.unregister(this.dispatcherID);
+
+    if (this.multilineIntervalID !== undefined)
+      clearInterval(this.multilineIntervalID);
+
+    if (this.customDispatcherID)
+      deregisterCustomDispatchOverridingHandler(this.customDispatcherID);
+  }
+
+  // navigates up/down through a multiline expression
+  // returns false or undefined if successful
+  // returns true if at the start/end of a multiline expression
+  doMultilineVerticalNav(key: "Up" | "Down" | "Shift-Up" | "Shift-Down") {
+    const up = key === "Up" || key === "Shift-Up";
+    const select = key.startsWith("Shift");
+
+    const focusedmq = MathQuillView.getFocusedMathquill();
+
+    let i = 0;
+    let linesPassed = 0;
+
+    const arrowdir = (select ? "Shift-" : "") + (up ? "Left" : "Right");
+    const oppositeArrowdir =
+      (select ? "Shift-" : "") + (!up ? "Left" : "Right");
+
+    // focus the mq element that was focused before hitting up/down
+    focusmq(focusedmq);
+
+    let nextFromBefore: Element | undefined | null;
+
+    // no need to do anything if there's no focused mathquill input
+    // return true to make sure it does normal behavior
+    if (!focusedmq) return true;
+
+    // get the original cursor horizontal position
+    // so we can snap to it later
+    const cursor = document.querySelector(".dcg-mq-cursor");
+    const originalCursorX =
+      this.lastRememberedCursorX ?? cursor?.getBoundingClientRect().left ?? 0;
+    const cursorPositions: number[] = [];
+
+    const ctrlr = getController(focusedmq);
+    // @ts-expect-error domfrag exists
+    const domfragProto = Object.getPrototypeOf(ctrlr.cursor.domFrag());
```

Comment:
Please add `domFrag` to the type in `desmosComponents`

Reply:
fixed!